docs/blog/things-to-know-when-developing-with-lit-element/index.md

```diff
@@ -0,0 +1,233 @@
+
+By falling back in the setter as opposed to the getter you can leverage the capabilities of a class component to prevent the render lifecycle to be started altogether. In the following code, no matter how the value of `this.closeDelay` gets to `300` the call to `this.requestUpdate()` is gated as long as the final value doesn't change.
+
+```js
+class MyThing extends LitElement {
+  @property({type: Number})
+  get closeDelay(): number {
+    return this._closeDelay;
+  }
+  set closeDelay(value: number) {
+    const closeDelay = value ?? 300;
+    if (closeDelay === this.closeDelay) return;
+    this.requestUpdate('closeDelay', this.closeDelay);
+    this._closeDelay = closeDelay;
+  }
+  _closeDelay = 300;
+...
+```
+
+Here you may still ask, "but, why it is so much more code?" and, in our ecosystem of "less is more", both from a DX as well as a UX/performance standpoint, it's a great question. It's more code because it is also a different level/type of capability. Here we get a default to our property, clear gating on the render lifecycle, and on top of that, we get a value that is held state-fully within a class that defines a DOM element. This means that not only can it take part in the render pipeline of the element that owns it, but that it is available for other elements that share its DOM tree to query as a container for that state. Not every application is architected with a want or need for this capability. Not every component is going to be leveraged at the scale where the checking is needed to confirm that the render lifecycle has no side effects in a way that benefits extra prevention of the lifecycle altogether. However, when you do, you might take a look at LitElement as a path towards attaining these capabilities.
+
+## So you like Typescript?
+
+Typescript loves a good `"Property is not definitely assigned in the constructor"` warning, and if you like Typescript enough to use it a lot, you'll likely run into it at some point. It's telling you this, because in contrast to what you've been told Typescript is NOT smarter than you and it can't tell if something _should_ always be available, only if it _might_ not be available, sometime. You can set an initial value to a property, and it'll never yell:
+
+```js
+class MyThing extends LitElement {
+  myProperty = 'string'; // always available, always a string
+}
+```
+
+If you're looking for that to _have_ to be initialized to use your element, so _you_ KNOW it's going to always have a value, but you want the consumer to initialize it, then you can tell Typescript that being the `!` operator once and be done with it:
```

Comment:
"then you can tell Typescript that being the `!` operator" -> "then you can tell Typescript that by using the `!` operator"